File: Esame_28/PYTHON/Auto.py
from Veicolo import Veicolo


class Auto(Veicolo):
    # COSTRUTTORE
    def __init__(self, codiceVeicolo, tipoVeicolo, targaVeicolo, modelloVeicolo, marcaVeicolo,
                 cilindrata, diesel):
        super().__init__(codiceVeicolo, tipoVeicolo, targaVeicolo, modelloVeicolo, marcaVeicolo)
        self._cilindrata = cilindrata
        self._diesel = diesel

    # TOSTRING PUNTO 3
    def toString(self):
        return super().toString() + str(self._cilindrata) + "\t" + str(self._diesel) + "\t-\t-\t"

File: Esame_28/JAVA/Auto.java

public class Auto extends Veicolo{
	private int cilindrata;
	private boolean diesel;
	
	public Auto(int codiceVeicolo, String tipoVeicolo, String targaVeicolo, String modelloVeicolo,String marcaVeicolo,
			int cilindrata, boolean diesel)
	{
		super(codiceVeicolo, tipoVeicolo, targaVeicolo, modelloVeicolo, marcaVeicolo);
		this.cilindrata = cilindrata;
		this.diesel = diesel;
	}
	//metodi
	public String toString()
	{
		return super.toString() + this.cilindrata +"\t"+ this.diesel +"\t-\t-\t";
	}
}
